frameworks/AutoGluon/exec.py

```diff
@@ -68,14 +69,21 @@ def run(dataset, config):
     # Persist model in memory that is going to be predicting to get correct inference latency
     predictor.persist_models('best')
 
+    def inference_time_classification(path: str):
+        data = TabularDataset(path)
+        return None, predictor.predict_proba(data, as_multiclass=True)
+
+    def inference_time_regression(path: str):
+        data = TabularDataset(path)
+        return predictor.predict(data, as_pandas=False), None
+
+    infer = inference_time_classification if is_classification else inference_time_regression
```

Comment:
If we are loading the test data inside the function, then we can save memory by not loading the test data in line 68 (otherwise it is existing in memory twice, which isn't optimal). Instead do `test_data = TabularDataset(test_path)` after the inference times are calculated to minimize memory usage. (after line 86)